hs_core/tests/api/native/test_bagit.py

```diff
@@ -0,0 +1,72 @@
+from unittest import TestCase
+from mock import patch
+
+from django.contrib.auth.models import Group, User
+
+from hs_core import hydroshare
+from hs_core.hydroshare import hs_bagit
+from hs_core.models import GenericResource
+
+from django_irods.storage import IrodsStorage
+
+
+class TestBagIt(TestCase):
+    def setUp(self):
+        # stop mocking of iRODS interaction in case those patchers are active
+        patch.stopall()
+        self.hs_group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+        # create a user
+        self.user = hydroshare.create_account(
+            'test_user@email.com',
+            username='mytestuser',
+            first_name='some_first_name',
+            last_name='some_last_name',
+            superuser=False,
+            groups=[self.hs_group]
+        )
+
+        # note: creating a resource calls the hs_bagit.create_bag() api
+        self.test_res = hydroshare.create_resource(
+            'GenericResource',
+            self.user,
+            'My Test Resource'
+            )
+
+    def tearDown(self):
+        User.objects.all().delete()
+        Group.objects.all().delete()
+        GenericResource.objects.all().delete()
+
+    def test_create_bag(self):
+        # the resource should have only one bags object
+        self.assertEquals(self.test_res.bags.count(), 1)
+        bag_first = self.test_res.bags.all().first()
+
+        # this is the api call we are testing
+        bag_second = hs_bagit.create_bag(self.test_res)
+
+        # resource should have 2 bags objects
+        self.assertEquals(self.test_res.bags.count(), 2)
+        self.assertIn(bag_second, self.test_res.bags.all())
+        self.assertIn(bag_first, self.test_res.bags.all())
+        self.assertNotEquals(bag_first, bag_second)
```

Comment:
@pkdash I am wondering whether we should allow resources to have more than 1 bags. In current hydroshare resoure creation workflow, create_bag() is only called by create_resource(), so only one bag is created for a resource. I am guessing the second create_bag() call does not really create another bag zip file on irods back end since resource modified flag is false so irods on-demand bag creation will not recreate another same bag, but in Django database, two Bags objects are created. This will not create any issues in our current HydroShare implementation, but does reveal something that should be better handled. I am thinking perhaps we should not allow a different Bags object being created, but rather make it point to the same Bags object if it already exists for the resource. If we modify create_bag() function to do that, this test needs to be changed as well. If you agree we should not allow multiple bags being created for the same resource, could you modify create_bag() function and this test_create_bag() accordingly to ensure only one Bags object to be created for each resource?